src/java/org/apache/nutch/segment/SegmentMerger.java

```diff
@@ -174,47 +174,51 @@
           // ignore
         }
       }
-      final SequenceFileRecordReader<Text, Writable> splitReader = new SequenceFileRecordReader<>(
-          job, (FileSplit) split);
+      final SequenceFileRecordReader<Text, Writable> splitReader = new SequenceFileRecordReader<>();
 
-      try {
-        return new SequenceFileRecordReader<Text, MetaWrapper>(job, fSplit) {
+      return new SequenceFileRecordReader<Text, MetaWrapper>() {
+
+        public MetaWrapper wrapper;
 
-          public synchronized boolean next(Text key, MetaWrapper wrapper)
-              throws IOException {
-            LOG.debug("Running OIF.next()");
+        @Override
+        public synchronized boolean nextKeyValue()
+            throws IOException, InterruptedException {
+          try {
+            LOG.debug("Running OIF.nextKeyValue()");
 
-            boolean res = splitReader.next(key, w);
+            splitReader.initialize(split, context);
+            this.initialize(split,context);
+            boolean res = splitReader.nextKeyValue();
+            wrapper = this.getCurrentValue();
```

Comment:
Hi @lewismc I think the TestSegmentMerger is failing here with a NullPointerException and I think that it is because the sequenceFileRecordReader here is not being intialized as expected. Please correct me if I am wrong. Thanks.